05-assistive-chatbot/chatbot/ingest/text_splitter.py

```diff
@@ -0,0 +1,47 @@
+import logging
+
+from langchain.docstore.document import Document
+from langchain_text_splitters import NLTKTextSplitter, RecursiveCharacterTextSplitter, SpacyTextSplitter
+
+logger = logging.getLogger(__name__)
+
+
+class TextSplitter:
+    def __init__(self, llm_client, token_limit, text_splitter_name, **text_splitter_args):
+        """
+        - llm_client is used to get the number of tokens in a text
+        - token_limit is the maximum number of tokens allowed by the embedding model
+        """
+        self.llm_client = llm_client
+        self.token_limit = token_limit
+        self.text_splitter = self.create_text_splitter(text_splitter_name, **text_splitter_args)
+
+    def create_text_splitter(self, choice, **kwargs):
+        logger.info("Creating %s", choice)
+        if choice == "NLTKTextSplitter":
+            logger.warning("  Not using arguments: %s", kwargs)
+            splitter = NLTKTextSplitter()
+        elif choice == "SpacyTextSplitter":
+            logger.warning("  Not using arguments: %s", kwargs)
+            splitter = SpacyTextSplitter()
+        elif choice == "RecursiveCharacterTextSplitter":
+            logger.info("  Using arguments: %s", kwargs)
+            splitter = RecursiveCharacterTextSplitter(
+                chunk_size=kwargs["chunk_size"], chunk_overlap=kwargs["chunk_overlap"]
+            )
+        return splitter
```

Comment:
I think this will throw an undefined error if `choice` doesn't match one of the conditions above; could explicitly branch on that and log a clearer error.

Reply:
Added `else` block